design/one-pager-component-mutable-and-versioning.md

```diff
@@ -0,0 +1,483 @@
+# Component mutable and versioning mechanism
+
+* Owner: Lei Zhang (@resouer), Jianbo Sun (@wonderflow)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+
+## Background
+
+OAM v1alpha1 made a requirement that developer need to define another ComponentSchematic with
+different name for an update (i.e. ComponentSchematic is immutable). We've got lots of feedback from
+real practices in Alibaba as well as [AWS ECS for OAM](https://github.com/awslabs/amazon-ecs-for-open-application-model#upgrade-and-scale-oam-workloads-with-oam-ecs)
+that it's not a good experience. Today [Component is mutable in v1alpha2 spec](https://github.com/oam-dev/spec/pull/356),
+this proposal is going to make it into implementation.
+
+Since component is becoming mutable, a very common question is how will the component versioning mechanism work.
+One of the requirement is the versioning mechanism should allow users to specify a fixed version of component
+so that the Application can be upgraded as before. On the other hand, the versioning mechanism should also be used
+by traits. For example, a trait can specify different revision and control deployment process. 
+
+
+## Goals
+
+1. Component is mutable. Developer who modify the Component object will directly trigger further
+upgrade of the application by default. 
+2. Component Version can be specified in ApplicationConfiguration. Operator can specify a fixed version
+of Component in ApplicationConfiguration. Component mutable won't affect Application directly in this case, 
+and can be upgraded in a more convenient way by working with trait.
+3. The versioning mechanism should compatible with current existing K8s workloads, eg, `K8s Deployment`, `OpenKruise`.
+  OAM should support two common ways used in community for versioning mechanism. 
+  - Workload itself can handle version. This means we can use `Deployment` only to do rollout or rollback without trait.
+  - Corporate with trait. OAM runtime should create different `Deployment` instances for different revision and let the trait do rollout or rollback.
```

Comment:
Your understand is correct. 

> it may exist simultaneously with the previous one

But only one of the case will happen at the same time. It's depending on the trait.